src/components/dbCalls.js

```diff
@@ -0,0 +1,435 @@
+    }
+    catch (error) {
+        console.error('Error fetching task: ', error);
+    }
+}
+
+//updates player's score
+const updatePointsForPlayer = async (player, points, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        const playerScore = parseInt(playerSnapshot.docs[0].data().score);
+        const newPoints = points + playerScore;
+        await updateDoc(playerdoc, { score: newPoints });
+    }
+    catch (error) {
+        console.error('Error updating player score: ', error);
+    }
+}
+
+//revives a player
+const updateIsAliveToTrueForPlayer = async (player, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        await updateDoc(playerdoc, { isAlive: true });
+    }
+    catch (error) {
+        console.error('Error reviving player: ', error);
+    }
+}
+
+//turns a player's alive status to false
+const updateIsAliveToFalseForPlayer = async (player, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        await updateDoc(playerdoc, { isAlive: false });
+    }
+    catch (error) {
+        console.error('Error killing player: ', error);
+    }
+}
+
+//marks task as completed
+const updateIsCompleteToTrueForTask = async (taskID, roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        const taskDocRef = doc(taskCollectionRef, taskID);
+        await updateDoc(taskDocRef, { isComplete: true });
+    }
+    catch (error) {
+        console.error('Error completing task: ', error);
+    }
+}
+
+//updates the 'completedBy' field of a task
+const updateCompletedByForTask = async (taskDocRef, checkedPlayers) => {
+    try {
+        await updateDoc(taskDocRef, { completedBy: checkedPlayers });
+    }
+    catch (error) {
+        console.error('Error updating completedBy: ', error);
+    }
+}
+
+const addTaskForRoom = async (task, roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        await addDoc(taskCollectionRef, task);
+    }
+    catch (error) {
+        console.error('Error adding task: ', error);
+    }
+}
+
+//check if task title already exists in database
+const checkForTaskDupesForRoom = async (task, roomID, createAlert) => {
```

Comment:
Database file should just handle database stuff, it shouldn't worry about creating alerts. Can we rewrite this just to check if there exists a task with a particular title and then do the proper error throwing/handling wherever this gets called?